Thunder/utils/canonical_files.py

```
@@ -0,0 +1,333 @@
+                await db.create_file_record(record)
+            _remember(record)
+            return record, stored_message, False
+        except DuplicateKeyError:
+            existing = await get_file_by_unique_id(file_unique_id)
+            if existing:
+                schedule_touch_file_record(existing, reused=True)
+                return existing, None, True
+            raise
+        except FloodWait:
+            raise
+        except Exception as e:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)